pandas/core/dtypes/cast.py

```diff
@@ -735,7 +745,15 @@ def astype_nansafe(arr, dtype, copy=True):
 
     if copy:
         return arr.astype(dtype, copy=True)
-    return arr.view(dtype)
+    else:
+        try:
+            return arr.view(dtype)
+        except TypeError:
+            if copy is None:
```

Comment:
why is this necessary? what breaks this?

Reply:
There was some object astype that failed (not sure if it's to or from object). I've reverted this change, we'll see what fails.

Reply:
```
In [10]: np.array([1, 2, 3], dtype='int64').view(object)
...
TypeError: Cannot change data-type for object array.
```

Reply:
Ah, yes. thanks. Pushed a simpler fix that hopefully works.